src/pgduckdb_metadata_cache.cpp

```diff
@@ -1,45 +1,157 @@
 extern "C" {
 #include "postgres.h"
 
+#include "access/htup_details.h"
+#include "catalog/dependency.h"
+#include "catalog/namespace.h"
+#include "catalog/pg_type.h"
+#include "catalog/pg_proc.h"
 #include "commands/extension.h"
+#include "nodes/bitmapset.h"
+#include "utils/builtins.h"
+#include "utils/catcache.h"
 #include "utils/inval.h"
+#include "utils/memutils.h"
 #include "utils/syscache.h"
 }
 
+#include "pgduckdb/vendor/pg_list.hpp"
+
 namespace pgduckdb {
 struct {
+	/*
+	 * Does the cache contain valid data, i.e. is it initialized? Or is it
+	 * stale and does it need to be refreshed? If this is false none of the
+	 * other fields should be read.
+	 */
 	bool valid;
+	/*
+	 * Is the pg_duckdb extension installed? If this is false all the other
+	 * fields (except valid) should be ignored. It is totally fine to have
+	 * valid=true and installed=false, this happens when the extension is not
+	 * installed and we cached that information.
+	 */
 	bool installed;
+	/* The Postgres OID of the pg_duckdb extension. */
+	Oid extension_oid;
+	/* A list of Postgres OIDs of functions that can only be executed by DuckDB */
+	List *duckdb_only_functions;
```

Comment:
Not mandatory but maybe use Bitmapset instead of List ?

Reply:
I did consider that, but `Oid` is `uint32` and Bitmapset can only store `int32`. So that doesn't work (afaict).

Reply:
Sure and as comment above - at this point list is going to be small to have cause any problems.

Reply:
I included a note about the datastructure choice in the codecomment now.